### DISCUSSION TITLE
IP pool is not assigned to the server when done from ui

### DISCUSSION THREAD

**Discussion by akhil** [postalserver/postal#1444] [Help with using Postal]
1. enable ip_pool in config file, 
2. made entries in the postal UI 
3. enabled in (checkbox) at org -> IPs -> IP Pools and server -> settings -> server settings UI
4. From `postal console` run ` Server.first.ip_pool` it shows `nil` 
5. ` select ip_pool_id from postal.servers;` shows `NULL`

To fix it I had to update the ip_pool_id using update query
` update postal.servers set ip_pool_id = 1;`

After the update statement the I can see the IPs getting rotated in the emails.

The UI flow is not updating the `server` table in `postal` schema.

**Comment by willpower232** [comment]
Did you restart Postal after changing the config file?

I don't think this has been reported before so must be something specific to your server.

**Comment by akhil** [comment]
> Did you restart Postal after changing the config file?
> 
> I don't think this has been reported before so must be something specific to your server.

I did. I was able to see the IP pool option in UI

**Comment by willpower232** [comment]
Ah of course, very odd.

Are you able to see anything that looks like a problem in the logs?

**Comment by akhil** [comment]
> Ah of course, very odd.
> 
> Are you able to see anything that looks like a problem in the logs?

No. I did a new installation using 18.04 script. (had to make local modification to get the rabbitmq deps to work)

**Comment by willpower232** [comment]
Are you able to replicate the problem at all?

**Comment by akhil** [comment]
> Are you able to replicate the problem at all?

Yes consistently.

**Comment by donatid** [comment]
Hello,

I also had the same issue, my outbound email always uses the interface's main ip address.
Here is the complete sql query and the results:

mysql> select ip_pool_id from postal.servers;
+------------+
| ip_pool_id |
+------------+
|       NULL |
|          1 |
|       NULL |
|       NULL |
|       NULL |
|       NULL |
|          1 |
|          1 |
+------------+
8 rows in set (0.00 sec)

mysql> update postal.servers set ip_pool_id = 1;
Query OK, 5 rows affected (0.03 sec)
Rows matched: 8  Changed: 5  Warnings: 0

mysql> select ip_pool_id from postal.servers;
+------------+
| ip_pool_id |
+------------+
|          1 |
|          1 |
|          1 |
|          1 |
|          1 |
|          1 |
|          1 |
|          1 |
+------------+
8 rows in set (0.00 sec)

I can confirm that IP POOL is working after I ran a query update in sql.

**Comment by willpower232** [comment]
Github isn't showing the link but #1622 suggests that the UI has a problem saving the value
